autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/ExtraTreesPreprocessorClassification.py

```diff
@@ -0,0 +1,157 @@
+from typing import Any, Dict, Optional, Union
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import (
+    CategoricalHyperparameter,
+    UniformFloatHyperparameter,
+    UniformIntegerHyperparameter,
+)
+
+import numpy as np
+
+from sklearn.base import BaseEstimator
+from sklearn.ensemble import ExtraTreesClassifier
+from sklearn.feature_selection import SelectFromModel
+
+from autoPyTorch.datasets.base_dataset import BaseDatasetPropertiesType
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.feature_preprocessing. \
+    base_feature_preprocessor import autoPyTorchFeaturePreprocessingComponent
+from autoPyTorch.utils.common import HyperparameterSearchSpace, add_hyperparameter, check_none
+
+
+class ExtraTreesPreprocessorClassification(autoPyTorchFeaturePreprocessingComponent):
+    """
+    Selects features based on importance weights calculated using extra trees
+    """
+    def __init__(self, bootstrap: bool = True, n_estimators: int = 10,
+                 criterion: str = "gini", max_features: float = 0.5,
+                 max_depth: Optional[Union[str, int]] = 5, min_samples_split: int = 2,
+                 min_samples_leaf: int = 1, min_weight_fraction_leaf: float = 0,
+                 max_leaf_nodes: Optional[Union[str, int]] = "none",
```

Comment:
Setting max_depth to None when we pass it to the classifier allows us to follow the default in sklearns documentation for this classifier. However, it should still be possible to pass a search space update where we either search for max_depth or set it to a constant value. These values come from the `get_hyperparameter_search_space` of the component and is passed to the component through `set_hyperparameters` method. Passing the python literal `None` there raises an error. Therefore we use a substitute which is "None" or "none" to indicate we dont want to search it. I'll add this to the documentation of this class to make it clearer there.

Reply:
Ok, I got what you mean although it is kind of problematic because it will mess up typing...
But then one suggestion, how about the following?:
1. Define the following in somewhere like in util
```
NoneType_ = Optional[str]  # This typing is exclusively for Literal["none", "None", None]
# NoneType_ = Optional[Literal["none", "None"]]  # For python 3.8 or later, we can use it.
```
2. Use `NoneType_` as a typing so that we can easily see the meaning
```
Optional[Union[str, int]]  ---> Union[int, NoneType_]
```

Reply:
we should do this. It seems elegent

Reply:
btw, we dont support python less than 3.8 so it should be fine

Reply:
I totally forgot we support 3.7 as well.